sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponse.java

```diff
@@ -4,34 +4,67 @@
 package com.generic.core.http.okhttp.implementation;
 
 import com.generic.core.http.models.HttpRequest;
+import com.generic.core.http.models.HttpResponse;
 import com.generic.core.models.BinaryData;
-import okhttp3.Response;
+import com.generic.core.models.HeaderName;
+import com.generic.core.models.Headers;
 import okhttp3.ResponseBody;
 
 /**
- * Default HTTP response for OkHttp.
+ * Base response class for OkHttp with implementations for response metadata.
  */
-public final class OkHttpResponse extends OkHttpResponseBase {
-    // Previously, this was 4096, but it is being changed to 8192 as that more closely aligns to what Netty uses as a
-    // default and will reduce the number of small allocations we'll need to make.
-    private static final int BYTE_BUFFER_CHUNK_SIZE = 8192;
+public class OkHttpResponse extends HttpResponse<BinaryData> {
+    private static final BinaryData EMPTY_BODY = BinaryData.fromBytes(new byte[0]);
 
     private final ResponseBody responseBody;
 
-    public OkHttpResponse(Response response, HttpRequest request, boolean eagerlyConvertHeaders) {
-        super(response, request, eagerlyConvertHeaders);
-
+    public OkHttpResponse(okhttp3.Response response, HttpRequest request, boolean eagerlyConvertHeaders,
+                          byte[] bodyBytes) {
+        super(request,
+            response.code(),
+            eagerlyConvertHeaders
+                ? fromOkHttpHeaders(response.headers())
+                : new OkHttpToAzureCoreHttpHeadersWrapper(response.headers()),
+            bodyBytes == null
+                ? response.body() == null
+                    ? EMPTY_BODY
+                    : BinaryData.fromStream(response.body().byteStream())
+                : BinaryData.fromBytes(bodyBytes));
         // innerResponse.body() getter will not return null for server returned responses.
         // It can be null:
-        // [a]. if response is built manually with null body (e.g for mocking)
+        // [a]. if response is built manually with null body (e.g. for mocking)
         // [b]. for the cases described here
         // [ref](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/).
         this.responseBody = response.body();
     }
 
-    @Override
-    public BinaryData getBody() {
-        return BinaryData.fromStream(this.responseBody.byteStream());
+    /**
+     * Creates {@link Headers Generic Core's headers} from {@link okhttp3.Headers OkHttp headers}.
+     *
+     * @param okHttpHeaders {@link okhttp3.Headers OkHttp headers}.
+     *
+     * @return {@link Headers Generic Core's headers}.
+     */
+    static Headers fromOkHttpHeaders(okhttp3.Headers okHttpHeaders) {
```

Comment:
Add back the deprecation warning. 
So that this can be done in a follow up PR.

Reply:
We don't need it, we're not using a deprecated API.